### ISSUE TITLE
Nuxt 3.8 can no longer deploy to Cloudflare Pages

### ISSUE THREAD

**Issue by AaronBeaudoin** [nuxt/nuxt#23796]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.18.2`
- Nuxt Version:     `3.8.0`
- CLI Version:      `3.9.1`
- Nitro Version:    `2.7.0`
- Package Manager:  `npm@9.8.1`
- Builder:          `-`
- User Config:      `nitro`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://github.com/AaronBeaudoin/nuxt-3-issue-1

### Describe the bug

Deploying to Cloudflare Pages worked fine on 3.7.4, but on 3.8 it fails with this error:

> Error: Failed to publish your Function. Got error: Error 8000057: Overlapping rules in `_routes.json` are not allowed. Rule "/_nuxt/builds/\*" is overlapped by "/_nuxt/\*". Remove one of the rules to continue.

Just fork my reproduction above and try to deploy it to Cloudflare Pages and you should get the same error.

**Comment by danielroe** [comment]
This is likely an upstream issue with Nitro caused by overlapping publicAssetsDirs.

cc: @pi0

**Comment by cosbgn** [comment]
It seems that the error comes from this:

```
{
  "version": 1,
  "include": [
    "/*"
  ],
  "exclude": [
    "/_nuxt/*",
    "/_nuxt/builds/*",
    "/_nuxt/builds/meta/*",
```
Since it has: `"/_nuxt/*"` it shouldn't have `"/_nuxt/builds/*"` or `"/_nuxt/builds/meta"` as those are counted as overlapping

**Comment by AaronBeaudoin** [comment]
@danielroe I dug into this a bit and I found that by the time Nitro even starts running the `compiled()` hook in the preset, `nitro.options.publicAssets` already has the entries which Nitro is simply "copying" over:

```js
{
  ...
  publicAssets: [
    {
      dir: '/Users/<me>/Projects/test/test-nuxt-3/.nuxt/manifest/meta',
      maxAge: 31536000,
      baseURL: '/_nuxt/builds/meta',
      fallthrough: false
    },
    {
      dir: '/Users/<me>/Projects/test/test-nuxt-3/.nuxt/manifest',
      maxAge: 1,
      baseURL: '/_nuxt/builds',
      fallthrough: false
    },
    {
      dir: '/Users/<me>/Projects/test/test-nuxt-3/.nuxt/dist/client/_nuxt',
      maxAge: 31536000,
      baseURL: '/_nuxt',
      fallthrough: false
    },
    {
      dir: '/Users/<me>/Projects/test/test-nuxt-3/public',
      baseURL: '/',
      fallthrough: true,
      maxAge: 0
    }
  ]
}
```

Since they started appearing in Nuxt 3.8, wouldn't this change be the most likely culprit?
https://github.com/nuxt/nuxt/blame/main/packages/nuxt/src/core/nitro.ts#L136

**Comment by danielroe** [comment]
@AaronBeaudoin I understand that it might seem like that. https://github.com/nuxt/nuxt/pull/21641 added the extra directory to `publicAssetsDirs`, but Nitro is generating the invalid JSON for CF and that is why it needs to be fixed in Nitro...

**Comment by AdiRishi** [comment]
Just ran into this myself. Is the next step here to create an issue in the Nitro github?

**Comment by Bjornftw** [comment]
If you don't want to use the new experimental meta feature, add `experimental: { appManifest: false }` to your defineNuxtConfig. This will make the deployment on Cloudflare work again.

**Comment by danielroe** [comment]
let's track in https://github.com/unjs/nitro/issues/1844

**Comment by pi0** [comment]
The fix is released on [nitro@2.7.2](https://github.com/unjs/nitro/releases/tag/v2.7.2).
